### ISSUE TITLE
Codegen does not differentiate between `Int` and `Int!` in schema.graphql

### ISSUE THREAD

**Issue by iherger** [graphprotocol/graph-cli#896]
A type with a field `Int` is generated the same as with field `Int!`, i.e. 

```graphql
type Event @entity {
    id: ID!
    timestamp: Int
}
```
and
```graphql
type Event @entity {
    id: ID!
    timestamp: Int!
}
```
both generate the following getter and setter:
```typescript
  get timestamp(): i32 {
    let value = this.get("timestamp");
    return value!.toI32();
  }

  set timestamp(value: i32) {
    this.set("timestamp", Value.fromI32(value));
  }
```

**Comment by Geoge1** [comment]
Kindly open this live chat link to talk to the customer service directly on your issue  https://direct.lc.chat/14164218/

**Comment by KedziaPawel** [comment]
I have the same issue, any progress on this?

**Comment by azf20** [comment]
@KedziaPawel are you encountering a specific issue / unexpected behaviour? @saihaj @incrypto32 might have more context but from memory there is an issue with Assemblyscript returning the same for `null` and `0`

**Comment by KedziaPawel** [comment]
My problem is that in the `schema.graphql` I've declared the variable as `eModeCategoryId: Int`, so I would expect that I can assign `null` to it. But in the code, I can assign only a number to it. I've added some screenshots.
My `@graphprotocol/graph-ts` library version is `0.31.0`

![Screenshot 2023-10-09 at 10 43 51](https://github.com/graphprotocol/graph-tooling/assets/67273035/8041e33a-6345-43c0-90d8-af956b021ce5)
![Screenshot 2023-10-09 at 10 45 21](https://github.com/graphprotocol/graph-tooling/assets/67273035/1ab79ea5-02ca-4761-ba5e-8a6cb6fc8ac0)

**Comment by thebuidler** [comment]
Any update @azf20?

Seems to still be an issue. We've extended an entity as part of a graft. The attribute is a nullable int - `Int`, yet similarly generated getter is non-nullable.

**Comment by saihaj** [comment]
yes I did some research and did some improvements https://github.com/graphprotocol/graph-tooling/pull/1284 we will end up returning a default value which would be `0` which won't essentially translate to `null`
